components/webdriver_server/lib.rs

```diff
@@ -347,6 +410,131 @@ impl<'de> Visitor<'de> for TupleVecMapVisitor {
     }
 }
 
+fn compute_tick_duration(tick_actions: &ActionSequence) -> u64 {
+    let mut duration = 0;
+    match &tick_actions.actions {
+        ActionsType::Null { actions } => {
+            for action in actions.iter() {
+                let NullActionItem::General(GeneralAction::Pause(pause_action)) = action;
+                duration = cmp::max(duration, pause_action.duration.unwrap_or(0));
+            }
+        },
+        ActionsType::Pointer {
+            parameters: _,
+            actions,
+        } => {
+            for action in actions.iter() {
+                if let PointerActionItem::General(GeneralAction::Pause(pause_action)) = action {
+                    duration = cmp::max(duration, pause_action.duration.unwrap_or(0));
+                } else if let PointerActionItem::Pointer(PointerAction::Move(move_action)) = action
+                {
+                    duration = cmp::max(duration, move_action.duration.unwrap_or(0));
+                }
+            }
+        },
+        _ => (),
+    }
+    duration
+}
+
+// https://w3c.github.io/webdriver/#dfn-dispatch-tick-actions
+fn dispatch_tick_actions(
+    session: &mut WebDriverSession,
+    tick_actions: &ActionSequence,
+    tick_duration: u64,
+) -> WebDriverResult<WebDriverResponse> {
+    // Step 1.1
+    let source_id = tick_actions.id.as_ref().unwrap();
```

Comment:
When can this be None?

Reply:
Based on [this](https://w3c.github.io/webdriver/#dfn-process-an-input-source-action-sequence) we should return an error if the id is either `undefined` or not a `string`. This check would make more sense in `webdriver-rust` than here, I'm not sure why they chose to make the id an `Optional`.

Reply:
Could you file a bug in https://bugzilla.mozilla.org/enter_bug.cgi?product=Testing&component=geckodriver about that?

Reply:
Yes, sure!

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1568327